plasma_framework/contracts/poc/fast_exits/Liquidity.sol

```diff
@@ -0,0 +1,186 @@
+        = PaymentTransactionModel.decode(rlpOutputTxToContract);
+        require(
+            decodedSecondTx.inputs[0] == bytes32(utxoPosInput),
+            "Wrong utxoPosInput provided"
+        );
+
+        verifyTxValidity(
+            utxoPosInput,
+            rlpInputCreationTx,
+            inputCreationTxInclusionProof
+        );
+        runExit(
+            utxoPosToExit,
+            rlpOutputTxToContract,
+            outputTxToContractInclusionProof
+        );
+
+        // store the resultant exitid as a trait for the nft and map it to the msg.sender
+        uint160 exitId = paymentExitGame.getStandardExitId(false, rlpOutputTxToContract, utxoPosToExit);
+        userExitIds[msg.sender].push(exitId);
+        exitIdtoUser[exitId] = msg.sender;
+
+        // associate the amount exiting to the exitId
+
+        FungibleTokenOutputModel.Output memory outputFromSecondTransaction
+        = decodedSecondTx.outputs[0];
+        uint256 amount = outputFromSecondTransaction.amount;
+        exitIdtoAmount[exitId] = amount;
+    }
+
+    /**
+     * @notice Check if the person calling is the same person who created the tx to the contract
+     * @param rlpInputCreationTx RLP-encoded first transaction that transfers to this contract
+     * @param utxoPosInput position of the output that created the inputs for second transaction
+    */
+    function verifyOwnership(
+        bytes memory rlpInputCreationTx,
+        uint256 utxoPosInput
+    ) internal {
+
+        PaymentTransactionModel.Transaction memory decodedFirstTx
+        = PaymentTransactionModel.decode(rlpInputCreationTx);
+        PosLib.Position memory position = PosLib.decode(utxoPosInput);
+        uint16 firstTransactionOutputIndex = position.outputIndex;
+
+        FungibleTokenOutputModel.Output memory outputFromFirstTransaction
+        = decodedFirstTx.outputs[firstTransactionOutputIndex];
+        address ownerFirstTxOutput = PaymentTransactionModel.getOutputOwner(outputFromFirstTransaction);
+        require(
+            ownerFirstTxOutput == msg.sender,
+            "Was not called by the first Tx owner"
+        );
+    }
+
+    /**
+     * @notice Verify the First Tx provided is valid
+     * @param utxoPosInput position of the output that created the inputs for second transaction
+     * @param rlpInputCreationTx RLP-encoded first transaction that transfers to this contract
+     * @param inputCreationTxInclusionProof First transactions inclusion proofs
+    */
+    function verifyTxValidity(
+        uint256 utxoPosInput,
+        bytes memory rlpInputCreationTx,
+        bytes memory inputCreationTxInclusionProof
+    ) internal {
+        PosLib.Position memory utxoDecoded = PosLib.decode(utxoPosInput);
+        utxoDecoded.outputIndex = 0;
+        (bytes32 root, ) = plasmaFrameworkInstance.blocks(utxoDecoded.blockNum);
+        require(root != bytes32(""), "Failed to get root of the block");
```

Comment:
is there any other way to compare empty `bytes32`?

Reply:
Umm, any idea?

Reply:
`bytes32(0x00)` should work and I think I prefer it this way

Reply:
Aah cool, thanks! :+1: I'll make it a point to switch to this in the next PR